app/(main)/league/all/page.tsx

```diff
@@ -3,23 +3,49 @@
 
 'use client';
 
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
 import { ENTRY_URL, LEAGUE_URL } from '@/const/global';
-import { getUserLeagues } from '@/utils/utils';
-import { ILeague } from '@/api/apiFunctions.interface';
+import { Button } from '@/components/Button/Button';
+import { getUserLeagues, fetchAvailableLeagues } from '@/utils/utils';
+import { ILeague, IAllLeagues } from '@/api/apiFunctions.interface';
+import { addUserToLeague } from '@/api/apiFunctions';
 import { LeagueCard } from '@/components/LeagueCard/LeagueCard';
 import { useDataStore } from '@/store/dataStore';
 import GlobalSpinner from '@/components/GlobalSpinner/GlobalSpinner';
 import React, { JSX, useEffect, useState } from 'react';
+import { toast } from 'react-hot-toast';
 
 /**
  * Renders the leagues component.
  * @returns {JSX.Element} The rendered leagues component.
  */
 const Leagues = (): JSX.Element => {
+  const [availableLeagues, setAvailableLeagues] = useState<IAllLeagues[]>([]);
   const [leagues, setLeagues] = useState<ILeague[]>([]);
   const [loadingData, setLoadingData] = useState<boolean>(true);
+  const [selectedLeagues, setSelectedLeagues] = useState<string | null>(null);
   const { user } = useDataStore((state) => state);
 
+  /**
+   * Fetches all available leagues.
+   * @param root0
+   * @param root0.participants
+   * @param root0.survivors
+   * @returns {Promise<void>}
+   */
+  const fetchLeagues = async (): Promise<void> => {
+    try {
+      const leagues = await fetchAvailableLeagues();
+      console.log(leagues);
```

Comment:
This probably broke your build because you cannot have console.log's in your code.